From db799fa56b02facb210f049bd0b00bae79280e0a Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 29 Jan 2016 19:49:26 +1100
Subject: [PATCH] d3d11: Fake ID3D11Device CheckFormatSupport return value

These values are based of an nvidia driver running windows 7.

Microsoft Matrix lookups.
https://msdn.microsoft.com/en-us/library/windows/desktop/mt427455%28v=vs.85%29.aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/mt427456%28v=vs.85%29.aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/mt426647%28v=vs.85%29.aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/mt426648%28v=vs.85%29.aspx

Fixes: https://bugs.winehq.org/show_bug.cgi?id=40046

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/d3d11/device.c | 681 +++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 680 insertions(+), 1 deletion(-)

diff --git a/dlls/d3d11/device.c b/dlls/d3d11/device.c
index 187b3f0..613c6ba 100644
--- a/dlls/d3d11/device.c
+++ b/dlls/d3d11/device.c
@@ -2698,11 +2698,690 @@ static HRESULT STDMETHODCALLTYPE d3d11_device_OpenSharedResource(ID3D11Device *i
     return E_NOTIMPL;
 }
 
+struct device_format_support
+{
+    DXGI_FORMAT format;
+    UINT supported;
+};
+
+static const struct device_format_support formats[] =
+{
+    {DXGI_FORMAT_UNKNOWN,        D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE },
+    {DXGI_FORMAT_R32G32B32A32_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R32G32B32A32_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_MIP_AUTOGEN | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_BLENDABLE | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32B32A32_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32B32A32_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32B32_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R32G32B32_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R32G32B32_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R32G32B32_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R16G16B16A16_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R16G16B16A16_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16B16A16_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16B16A16_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16B16A16_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16B16A16_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R32G32_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_MIP_AUTOGEN | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_BLENDABLE | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G32_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32G8X24_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_D32_FLOAT_S8X24_UINT,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_DEPTH_STENCIL | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET },
+    {DXGI_FORMAT_R32_FLOAT_X8X24_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE_COMPARISON |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_X32_TYPELESS_G8X24_UINT,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R10G10B10A2_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R10G10B10A2_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_R10G10B10A2_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R11G11B10_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8B8A8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R8G8B8A8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_R8G8B8A8_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_R8G8B8A8_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8B8A8_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8B8A8_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R16G16_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16G16_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_D32_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_DEPTH_STENCIL | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET },
+    {DXGI_FORMAT_R32_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE_COMPARISON |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_IA_INDEX_BUFFER | D3D11_FORMAT_SUPPORT_SO_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R32_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_SO_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R24G8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_D24_UNORM_S8_UINT,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_DEPTH_STENCIL | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET },
+    {DXGI_FORMAT_R24_UNORM_X8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE_COMPARISON |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_X24_TYPELESS_G8_UINT,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R8G8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R8G8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8G8_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R16_FLOAT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_D16_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_DEPTH_STENCIL | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET },
+    {DXGI_FORMAT_R16_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE_COMPARISON | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_MIP_AUTOGEN | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_IA_INDEX_BUFFER | D3D11_FORMAT_SUPPORT_TEXTURE1D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_RENDER_TARGET |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R16_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_R8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8_UINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8_SNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R8_SINT,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_A8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    /* DXGI_FORMAT_R1_UNORM(66) currently unsupported */
+    {DXGI_FORMAT_R9G9B9E5_SHAREDEXP,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE },
+    {DXGI_FORMAT_R8G8_B8G8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE },
+    {DXGI_FORMAT_G8R8_G8B8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE }, 
+    {DXGI_FORMAT_BC1_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT  },
+    {DXGI_FORMAT_BC1_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC1_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC2_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC2_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC2_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC3_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC3_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC3_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC4_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC4_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC4_SNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC5_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT }, 
+    {DXGI_FORMAT_BC5_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_BC5_SNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_TEXTURE3D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURECUBE | D3D11_FORMAT_SUPPORT_SHADER_LOAD |
+                                 D3D11_FORMAT_SUPPORT_SHADER_SAMPLE | D3D11_FORMAT_SUPPORT_MIP |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    /* DXGI_FORMAT_B5G6R5_UNORM   (85) currently unsupported */
+    /* DXGI_FORMAT_B5G5R5A1_UNORM (86) currently unsupported */
+    {DXGI_FORMAT_B8G8R8A8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_B8G8R8X8_UNORM,
+                                 D3D11_FORMAT_SUPPORT_BUFFER | D3D11_FORMAT_SUPPORT_IA_VERTEX_BUFFER |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD },
+    {DXGI_FORMAT_R10G10B10_XR_BIAS_A2_UNORM,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE2D | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_B8G8R8A8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_B8G8R8A8_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_DISPLAY | D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET | D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD |
+                                 D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST },
+    {DXGI_FORMAT_B8G8R8X8_TYPELESS,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_CPU_LOCKABLE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT },
+    {DXGI_FORMAT_B8G8R8X8_UNORM_SRGB,
+                                 D3D11_FORMAT_SUPPORT_TEXTURE1D | D3D11_FORMAT_SUPPORT_TEXTURE2D |
+                                 D3D11_FORMAT_SUPPORT_TEXTURE3D | D3D11_FORMAT_SUPPORT_TEXTURECUBE |
+                                 D3D11_FORMAT_SUPPORT_SHADER_LOAD | D3D11_FORMAT_SUPPORT_SHADER_SAMPLE |
+                                 D3D11_FORMAT_SUPPORT_MIP | D3D11_FORMAT_SUPPORT_MIP_AUTOGEN |
+                                 D3D11_FORMAT_SUPPORT_RENDER_TARGET | D3D11_FORMAT_SUPPORT_BLENDABLE |
+                                 D3D11_FORMAT_SUPPORT_CPU_LOCKABLE | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RESOLVE |
+                                 D3D11_FORMAT_SUPPORT_CAST_WITHIN_BIT_LAYOUT | D3D11_FORMAT_SUPPORT_MULTISAMPLE_RENDERTARGET |
+                                 D3D11_FORMAT_SUPPORT_MULTISAMPLE_LOAD }
+    /* DXGI_FORMAT_BC6H_TYPELESS  (94) currently unsupported */
+    /* DXGI_FORMAT_BC6H_UF16      (95) currently unsupported */
+    /* DXGI_FORMAT_BC6H_SF16      (96) currently unsupported */
+    /* DXGI_FORMAT_BC7_TYPELESS   (97) currently unsupported */
+    /* DXGI_FORMAT_BC7_UNORM      (98) currently unsupported */
+    /* DXGI_FORMAT_BC7_UNORM_SRGB (99) currently unsupported */
+    /* DXGI_FORMAT_AYUV           (100) currently unsupported */
+    /* DXGI_FORMAT_Y410           (101) currently unsupported */
+    /* DXGI_FORMAT_Y416           (102) currently unsupported */
+    /* DXGI_FORMAT_NV12           (103) currently unsupported */
+    /* DXGI_FORMAT_P010           (104) currently unsupported */
+    /* DXGI_FORMAT_P016           (105) currently unsupported */
+    /* DXGI_FORMAT_420_OPAQUE     (106) currently unsupported */
+    /* DXGI_FORMAT_YUY2           (107) currently unsupported */
+    /* DXGI_FORMAT_Y210           (108) currently unsupported */
+    /* DXGI_FORMAT_Y216           (109) currently unsupported */
+    /* DXGI_FORMAT_NV11           (110) currently unsupported */
+    /* DXGI_FORMAT_AI44           (111) currently unsupported */
+    /* DXGI_FORMAT_IA44           (112) currently unsupported */
+    /* DXGI_FORMAT_P8             (113) currently unsupported */
+    /* DXGI_FORMAT_A8P8           (114) currently unsupported */
+    /* DXGI_FORMAT_B4G4R4A4_UNORM (115) currently unsupported */
+};
+
 static HRESULT STDMETHODCALLTYPE d3d11_device_CheckFormatSupport(ID3D11Device *iface, DXGI_FORMAT format,
         UINT *format_support)
 {
-    FIXME("iface %p, format %u, format_support %p stub!\n", iface, format, format_support);
+    int i;
+
+    TRACE("iface %p, format %u, format_support %p semi-stub!\n", iface, format, format_support);
+
+    for (i = 0; i < (sizeof(formats) / sizeof(*formats)); ++i)
+    {
+        if(formats[i].format == format)
+        {
+            *format_support = formats[i].supported;
+            return S_OK;
+        }
+    }
 
+    FIXME("DXGI_FORMAT %u currently not supported\n", format);
     return E_NOTIMPL;
 }
 
-- 
1.9.1

